### PR TITLE
Mining fees info mechanism

### PR DESCRIPTION
**Pull request by martinmedina**

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 69 issues
* ![MAJOR][MAJOR] 36 major
* ![MINOR][MINOR] 13 minor
* ![INFO][INFO] 20 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [MinerServerImpl.java#L148](https://github.com/rsksmart/rskj/blob/37f79e360ac69c95cd92821029d316a25fcb671f/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L148): Method co.rsk.mine.MinerServerImpl.submitBitcoinBlock(String, BtcBlock) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L175](https://github.com/rsksmart/rskj/blob/37f79e360ac69c95cd92821029d316a25fcb671f/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L175): Method co.rsk.mine.MinerServerImpl.submitBitcoinBlock(String, BtcBlock) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L193](https://github.com/rsksmart/rskj/blob/37f79e360ac69c95cd92821029d316a25fcb671f/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L193): Method co.rsk.mine.MinerServerImpl.submitBitcoinBlock(String, BtcBlock) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L287](https://github.com/rsksmart/rskj/blob/37f79e360ac69c95cd92821029d316a25fcb671f/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L287): Boxing/unboxing to parse a primitive co.rsk.mine.MinerServerImpl.getWork() [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3ADM_BOXED_PRIMITIVE_FOR_PARSING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L287](https://github.com/rsksmart/rskj/blob/37f79e360ac69c95cd92821029d316a25fcb671f/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L287): Method co.rsk.mine.MinerServerImpl.getWork() converts String to primitive using excessive boxing [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ANAB_NEEDLESS_BOXING_PARSE)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L307](https://github.com/rsksmart/rskj/blob/37f79e360ac69c95cd92821029d316a25fcb671f/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L307): Method co.rsk.mine.MinerServerImpl.updateGetWork(Block, boolean) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L363](https://github.com/rsksmart/rskj/blob/37f79e360ac69c95cd92821029d316a25fcb671f/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L363): Method co.rsk.mine.MinerServerImpl.buildBlockToMine(Block, boolean) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L366](https://github.com/rsksmart/rskj/blob/37f79e360ac69c95cd92821029d316a25fcb671f/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L366): Method co.rsk.mine.MinerServerImpl.buildBlockToMine(Block, boolean) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L368](https://github.com/rsksmart/rskj/blob/37f79e360ac69c95cd92821029d316a25fcb671f/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L368): Method co.rsk.mine.MinerServerImpl.buildBlockToMine(Block, boolean) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L452](https://github.com/rsksmart/rskj/blob/37f79e360ac69c95cd92821029d316a25fcb671f/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L452): Method co.rsk.mine.MinerServerImpl$NewBlockListener.onBlock(Block, List) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 59 issues
* ![MAJOR][MAJOR] 26 major
* ![MINOR][MINOR] 13 minor
* ![INFO][INFO] 20 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [Remasc.java#L176](https://github.com/rsksmart/rskj/blob/c8e6b51fa667686d316960a1c1e60596f1a9e66d/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L176): Method co.rsk.remasc.Remasc.payWithSiblings(BlockHeader, BigInteger, List, boolean) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [RemascContract.java#L100](https://github.com/rsksmart/rskj/blob/c8e6b51fa667686d316960a1c1e60596f1a9e66d/rskj-core/src/main/java/co/rsk/remasc/RemascContract.java#L100): Method co.rsk.remasc.RemascContract.execute(byte[]) throws alternative exception from catch block without history [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALEST_LOST_EXCEPTION_STACK_TRACE)
1. ![MAJOR][MAJOR] [RemascContract.java#L109](https://github.com/rsksmart/rskj/blob/c8e6b51fa667686d316960a1c1e60596f1a9e66d/rskj-core/src/main/java/co/rsk/remasc/RemascContract.java#L109): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [RemascContract.java#L115](https://github.com/rsksmart/rskj/blob/c8e6b51fa667686d316960a1c1e60596f1a9e66d/rskj-core/src/main/java/co/rsk/remasc/RemascContract.java#L115): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Web3RskImpl.java#L71](https://github.com/rsksmart/rskj/blob/c8e6b51fa667686d316960a1c1e60596f1a9e66d/rskj-core/src/main/java/co/rsk/rpc/Web3RskImpl.java#L71): Method co.rsk.rpc.Web3RskImpl.mnr_submitBitcoinBlock(String) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [Web3RskImpl.java#L114](https://github.com/rsksmart/rskj/blob/c8e6b51fa667686d316960a1c1e60596f1a9e66d/rskj-core/src/main/java/co/rsk/rpc/Web3RskImpl.java#L114): Change this "try" to a try-with-resources. (sonar.java.source not set. Assuming 7 or greater.) [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2093)
1. ![MAJOR][MAJOR] [Web3RskImpl.java#L133](https://github.com/rsksmart/rskj/blob/c8e6b51fa667686d316960a1c1e60596f1a9e66d/rskj-core/src/main/java/co/rsk/rpc/Web3RskImpl.java#L133): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [Web3RskImpl.java#L144](https://github.com/rsksmart/rskj/blob/c8e6b51fa667686d316960a1c1e60596f1a9e66d/rskj-core/src/main/java/co/rsk/rpc/Web3RskImpl.java#L144): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [Web3RskImpl.java#L144](https://github.com/rsksmart/rskj/blob/c8e6b51fa667686d316960a1c1e60596f1a9e66d/rskj-core/src/main/java/co/rsk/rpc/Web3RskImpl.java#L144): Either remove or fill this block of code. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00108)
1. ![MAJOR][MAJOR] [TypeConverter.java#L30](https://github.com/rsksmart/rskj/blob/c8e6b51fa667686d316960a1c1e60596f1a9e66d/rskj-core/src/main/java/org/ethereum/rpc/TypeConverter.java#L30): Add a private constructor to hide the implicit public one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1118)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 56 issues
* ![MAJOR][MAJOR] 23 major
* ![MINOR][MINOR] 13 minor
* ![INFO][INFO] 20 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [RemascContract.java#L100](https://github.com/rsksmart/rskj/blob/a5e688c9fea81ab7e54970783b5425aacc53baec/rskj-core/src/main/java/co/rsk/remasc/RemascContract.java#L100): Method co.rsk.remasc.RemascContract.execute(byte[]) throws alternative exception from catch block without history [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALEST_LOST_EXCEPTION_STACK_TRACE)
1. ![MAJOR][MAJOR] [RemascContract.java#L109](https://github.com/rsksmart/rskj/blob/a5e688c9fea81ab7e54970783b5425aacc53baec/rskj-core/src/main/java/co/rsk/remasc/RemascContract.java#L109): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [RemascContract.java#L115](https://github.com/rsksmart/rskj/blob/a5e688c9fea81ab7e54970783b5425aacc53baec/rskj-core/src/main/java/co/rsk/remasc/RemascContract.java#L115): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Web3RskImpl.java#L114](https://github.com/rsksmart/rskj/blob/a5e688c9fea81ab7e54970783b5425aacc53baec/rskj-core/src/main/java/co/rsk/rpc/Web3RskImpl.java#L114): Change this "try" to a try-with-resources. (sonar.java.source not set. Assuming 7 or greater.) [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2093)
1. ![MAJOR][MAJOR] [Web3RskImpl.java#L133](https://github.com/rsksmart/rskj/blob/a5e688c9fea81ab7e54970783b5425aacc53baec/rskj-core/src/main/java/co/rsk/rpc/Web3RskImpl.java#L133): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [Web3RskImpl.java#L144](https://github.com/rsksmart/rskj/blob/a5e688c9fea81ab7e54970783b5425aacc53baec/rskj-core/src/main/java/co/rsk/rpc/Web3RskImpl.java#L144): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [Web3RskImpl.java#L144](https://github.com/rsksmart/rskj/blob/a5e688c9fea81ab7e54970783b5425aacc53baec/rskj-core/src/main/java/co/rsk/rpc/Web3RskImpl.java#L144): Either remove or fill this block of code. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00108)
1. ![MAJOR][MAJOR] [DataWord.java#L53](https://github.com/rsksmart/rskj/blob/a5e688c9fea81ab7e54970783b5425aacc53baec/rskj-core/src/main/java/org/ethereum/vm/DataWord.java#L53): Constructor new org.ethereum.vm.DataWord() makes call to non-final method [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APCOA_PARTIALLY_CONSTRUCTED_OBJECT_ACCESS)
1. ![MAJOR][MAJOR] [DataWord.java#L69](https://github.com/rsksmart/rskj/blob/a5e688c9fea81ab7e54970783b5425aacc53baec/rskj-core/src/main/java/org/ethereum/vm/DataWord.java#L69): Rename "data" which hides the field declared at line 50. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AHiddenFieldCheck)
1. ![MAJOR][MAJOR] [DataWord.java#L85](https://github.com/rsksmart/rskj/blob/a5e688c9fea81ab7e54970783b5425aacc53baec/rskj-core/src/main/java/org/ethereum/vm/DataWord.java#L85): Constructor new org.ethereum.vm.DataWord(byte[]) makes call to non-final method [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APCOA_PARTIALLY_CONSTRUCTED_OBJECT_ACCESS)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java [hunk 1]
```diff
@@ -156,8 +156,10 @@ public void submitBitcoinBlock(String blockHashForMergedMining, co.rsk.bitcoinj.
             Block workingBlock = blocksWaitingforPoW.get(key);
 
             if (workingBlock == null) {
-                logger.warn("Cannot publish block, could not find hash " + blockHashForMergedMining + " in the cache");
-                return;
+                String message = "Cannot publish block, could not find hash {} in the cache" + blockHashForMergedMining + " in the cache";
```
Comment (donequis) [inline rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java]: Weird message. It was OK

"Cannot publish block, could not find hash " + blockHashForMergedMining + " in the cache"

Comment (martinmedina) [reply rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java]: Fixed

**Comment by rskops** [comment]
SonarQube analysis reported 56 issues
* ![MAJOR][MAJOR] 23 major
* ![MINOR][MINOR] 13 minor
* ![INFO][INFO] 20 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [RemascContract.java#L100](https://github.com/rsksmart/rskj/blob/18706012d5db87e81def67960d047c165c55f22d/rskj-core/src/main/java/co/rsk/remasc/RemascContract.java#L100): Method co.rsk.remasc.RemascContract.execute(byte[]) throws alternative exception from catch block without history [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALEST_LOST_EXCEPTION_STACK_TRACE)
1. ![MAJOR][MAJOR] [RemascContract.java#L109](https://github.com/rsksmart/rskj/blob/18706012d5db87e81def67960d047c165c55f22d/rskj-core/src/main/java/co/rsk/remasc/RemascContract.java#L109): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [RemascContract.java#L115](https://github.com/rsksmart/rskj/blob/18706012d5db87e81def67960d047c165c55f22d/rskj-core/src/main/java/co/rsk/remasc/RemascContract.java#L115): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Web3RskImpl.java#L114](https://github.com/rsksmart/rskj/blob/18706012d5db87e81def67960d047c165c55f22d/rskj-core/src/main/java/co/rsk/rpc/Web3RskImpl.java#L114): Change this "try" to a try-with-resources. (sonar.java.source not set. Assuming 7 or greater.) [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2093)
1. ![MAJOR][MAJOR] [Web3RskImpl.java#L133](https://github.com/rsksmart/rskj/blob/18706012d5db87e81def67960d047c165c55f22d/rskj-core/src/main/java/co/rsk/rpc/Web3RskImpl.java#L133): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [Web3RskImpl.java#L144](https://github.com/rsksmart/rskj/blob/18706012d5db87e81def67960d047c165c55f22d/rskj-core/src/main/java/co/rsk/rpc/Web3RskImpl.java#L144): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [Web3RskImpl.java#L144](https://github.com/rsksmart/rskj/blob/18706012d5db87e81def67960d047c165c55f22d/rskj-core/src/main/java/co/rsk/rpc/Web3RskImpl.java#L144): Either remove or fill this block of code. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00108)
1. ![MAJOR][MAJOR] [DataWord.java#L53](https://github.com/rsksmart/rskj/blob/18706012d5db87e81def67960d047c165c55f22d/rskj-core/src/main/java/org/ethereum/vm/DataWord.java#L53): Constructor new org.ethereum.vm.DataWord() makes call to non-final method [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APCOA_PARTIALLY_CONSTRUCTED_OBJECT_ACCESS)
1. ![MAJOR][MAJOR] [DataWord.java#L69](https://github.com/rsksmart/rskj/blob/18706012d5db87e81def67960d047c165c55f22d/rskj-core/src/main/java/org/ethereum/vm/DataWord.java#L69): Rename "data" which hides the field declared at line 50. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AHiddenFieldCheck)
1. ![MAJOR][MAJOR] [DataWord.java#L85](https://github.com/rsksmart/rskj/blob/18706012d5db87e81def67960d047c165c55f22d/rskj-core/src/main/java/org/ethereum/vm/DataWord.java#L85): Constructor new org.ethereum.vm.DataWord(byte[]) makes call to non-final method [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APCOA_PARTIALLY_CONSTRUCTED_OBJECT_ACCESS)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/co/rsk/rpc/Web3RskImpl.java [hunk 2]
```diff
@@ -84,7 +86,13 @@ public void mnr_submitBitcoinBlock(String bitcoinBlockHex) {
         System.arraycopy(coinbaseAsByteArray, rskTagPosition+ RskMiningConstants.RSK_TAG.length, blockHashForMergedMiningArray, 0, blockHashForMergedMiningArray.length);
         String blockHashForMergedMining = TypeConverter.toJsonHex(blockHashForMergedMiningArray);
 
-        minerServer.submitBitcoinBlock(blockHashForMergedMining, bitcoinBlock);
+        SubmitBlockResult result = minerServer.submitBitcoinBlock(blockHashForMergedMining, bitcoinBlock);
+
+        if("OK".equals(result.getStatus())) {
```
Comment (donequis) [inline rskj-core/src/main/java/co/rsk/rpc/Web3RskImpl.java]: I don't like this strcmp

**Comment by donequis** [comment]
btw, this is being merged into master and is a hf

File: rskj-core/src/main/java/co/rsk/rpc/Web3RskImpl.java (hunk 2 above)
Comment (martinmedina) [reply rskj-core/src/main/java/co/rsk/rpc/Web3RskImpl.java]: It's only use is there, I don't think that there is a need to use an enum or smth like that.
What do you suggest ?

**Pull request merged**